src/main/java/com/thealgorithms/datastructures/buffers/DoubleBuffer.java

```diff
@@ -0,0 +1,105 @@
+package com.thealgorithms.datastructures.buffers;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The {@code DoubleBuffer} class implements a double buffering mechanism.
+ * It maintains two buffers, allowing the program to switch between them.
+ * This class is useful for separating read/write operations and optimizing memory usage.
+ *
+ * @param <Item> The type of elements stored in the double buffer.
+ */
+public class DoubleBuffer<Item> {
```

Comment:
What do you think about using `T` instead of `Item`. Because this is common practice